sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLQuerySuite.scala

```diff
@@ -2015,4 +2015,121 @@ class SQLQuerySuite extends QueryTest with SQLTestUtils with TestHiveSingleton {
       checkAnswer(table.filter($"p" === "p1\" and q=\"q1").select($"a"), Row(4))
     }
   }
+
+  test("TRIM function-BOTH") {
```

Comment:
These tests can also be put in sql/SQLQuerySuite. By the way, IIUC, the codeGen path is already tested in `StringExpressionSuite` by `checkEvaluation`, right?